src/ostree/ot-builtin-static-delta.c

```diff
@@ -438,13 +446,63 @@ ot_static_delta_builtin_apply_offline (int argc, char **argv, OstreeCommandInvoc
       return FALSE;
     }
 
+#if defined(HAVE_LIBSODIUM)
+  /* Initialize crypto system */
+  opt_sign_name = opt_sign_name ?: OSTREE_SIGN_NAME_ED25519;
+#endif
+
+  if (opt_sign_name)
+    {
+      sign = ostree_sign_get_by_name (opt_sign_name, error);
+      if (!sign)
+        {
+          g_print("Sign-type not supported\n");
+          return FALSE;
```

Comment:
In the "GError model" if `FALSE` is returned then the `error` must be set.

You want just:
`return glnx_throw (error, "Signing type %s is not supported", opt_sign_name);`